systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java

```diff
@@ -745,6 +751,138 @@ void testMultiNodeKafkaConnectWithConnectorCreation() {
+                    .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+                .endMetadata()
+                .withNewSpec()
+                    .withBootstrapServers(KafkaResources.tlsBootstrapAddress(CLUSTER_NAME))
+                    .withNewKafkaClientAuthenticationScramSha512()
+                        .withNewUsername(weirdUserName)
+                        .withPasswordSecret(new PasswordSecretSourceBuilder()
+                            .withSecretName(weirdUserName)
+                            .withPassword("password")
+                            .build())
+                    .endKafkaClientAuthenticationScramSha512()
+                    .addToConfig("key.converter.schemas.enable", false)
+                    .addToConfig("value.converter.schemas.enable", false)
+                    .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                    .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                    .withVersion(Environment.ST_KAFKA_VERSION)
+                    .withReplicas(1)
+                    .withNewTls()
+                        .withTrustedCertificates(new CertSecretSourceBuilder()
+                                .withCertificate("ca.crt")
+                                .withNewSecretName(KafkaResources.clusterCaCertificateSecretName(CLUSTER_NAME))
+                                .build())
+                    .endTls()
+                .endSpec()
+                .done();
+
+        testConnectAuthorizationWithWeirdUserName(weirdUserName, SecurityProtocol.SASL_SSL);
+    }
+
+    void testConnectAuthorizationWithWeirdUserName(String userName, SecurityProtocol securityProtocol) {
+        String topicName = TOPIC_NAME + rng.nextInt(Integer.MAX_VALUE);
+        KafkaTopicResource.topic(CLUSTER_NAME, topicName).done();
+        String connectorPodName = kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-connect").get(0).getMetadata().getName();
+
+        KafkaConnectorResource.kafkaConnector(CLUSTER_NAME)
+                .editSpec()
+                    .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                    .addToConfig("topics", topicName)
+                    .addToConfig("file", Constants.DEFAULT_SINK_FILE_PATH)
+                .endSpec().done();
+
+        BasicExternalKafkaClient basicExternalKafkaClient = new BasicExternalKafkaClient.Builder()
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(CLUSTER_NAME)
+                .withKafkaUsername(userName)
+                .withMessageCount(MESSAGE_COUNT)
+                .withConsumerGroupName(CONSUMER_GROUP_NAME + rng.nextInt(Integer.MAX_VALUE))
+                .withSecurityProtocol(securityProtocol)
+                .withTopicName(topicName)
+                .build();
+
+        assertThat(basicExternalKafkaClient.sendMessagesTls(), CoreMatchers.is(MESSAGE_COUNT));
```

Comment:
Please use only `is(.....)` without the prefix class. You will need to change the imports )